internal/hud/hud.go

```diff
@@ -83,6 +88,32 @@ func (h *Hud) Close() {
 	}
 }
 
+func (h *Hud) handleScreenEvent(ctx context.Context, st *store.Store, ev tcell.Event) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	switch ev := ev.(type) {
+	case *tcell.EventKey:
+		switch ev.Key() {
+		case tcell.KeyEscape:
+			h.Close()
+			h.r.Reset()
+		case tcell.KeyRune:
+			switch r := ev.Rune(); {
+			case r >= '1' && r <= '9':
+				st.Dispatch(NewShowErrorAction(int(r - '0')))
+			}
+		case tcell.KeyUp:
+			h.r.rty.ElementScroller("resources").UpElement()
+		case tcell.KeyDown:
+			h.r.rty.ElementScroller("resources").DownElement()
+		case tcell.KeyEnter:
+			activeItem := h.r.rty.ElementScroller("resources").GetSelectedIndex()
```

Comment:
i guess this is fine for now, but i'm a little bit nervous about how state is flowing thru this system, and how we're digging thru the component tree to read/write state